client/src/Context/AuthContext.tsx

```diff
@@ -0,0 +1,209 @@
+import { useState, createContext, useEffect, ReactNode } from "react";
+import { useNavigate } from "react-router-dom";
+import { getToken } from "../utils/getToken";
+
+interface AuthContextType {
+  newUser: SignUp;
+  selectedFile: File | string;
+  submitImage: (e: React.MouseEvent<HTMLButtonElement>) => void;
+  signUp: () => void;
+  logIn: () => void;
+  setSelectedFile: (selectedFile: File | string) => void;
+  setNewUser: (newUser: SignUp) => void;
+  user: boolean;
+  setUser: (user: boolean) => void;
+  loginUser: Login;
+  setLoginUser: (loginUser: Login) => void;
+  logOut: () => void;
+}
+interface Props {
+  children: ReactNode;
+}
+
+//TODO decide if array should include object id or something else, then change 'any';
+interface LoginResult {
+  accessToken: string;
+  isAuthenticated: boolean;
+  message: string;
+  user: {
+    email: string;
+    firstName: string;
+    image: string;
+    lastName: string;
+    starredCoworking: Array<any>;
+    starredMentorship: Array<any>;
+    starredShadowing: Array<any>;
+    username: string;
+    _id: number;
+  };
+}
+
+//TODO add image in respone?
+interface SignupResult {
+  accessToken: string;
+  message: string;
+  user: {
+    email: string;
+    firstName: string;
+    isLoggedin: boolean;
+    lastName: string;
+    starredCoworking: Array<any>;
+    starredMentorship: Array<any>;
+    starredShadowing: Array<any>;
+    username: string;
+    _id: number;
+  };
+}
+
+interface ImageResult {
+  image: string;
+  message: string;
+}
+
+//TODO display error messages
+
+export const AuthContext = createContext<AuthContextType>(undefined!);
+
+export const AuthContextProvider: React.FC<Props> = ({ children }) => {
+  const [user, setUser] = useState<boolean>(false);
+  const [selectedFile, setSelectedFile] = useState<string | File>("");
+  const [newUser, setNewUser] = useState<SignUp>({
+    firstName: "",
+    lastName: "",
+    username: "",
+    password: "",
+    email: "",
+    image: "",
+    isAdmin: false,
+  });
+  const [loginUser, setLoginUser] = useState<Login>({
```

Comment:
I am wondering if in the future we could extend this object to containing an entire user profile? Let's discuss this, it could allow us to minimise the number of user state objects.

Reply:
I did some changes to the interfaces, now it is more compact. But this above are the initial values, don't think we can optimise that

Reply:
ok found a mistake in password in the interface, already fixed it